Backend/Tumultu/Application/Common/Interfaces/IIdentityService.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Tumultu.Application.Common.Models;
+using Tumultu.Domain.Constants;
+using Tumultu.Domain.Entities;
+
+namespace Tumultu.Application.Common.Interfaces;
+
+public interface IIdentityService
+{
+    Task<User?> GetUserById(Guid id);
+    Task<bool> IsInRoleAsync(Guid? id, Roles role);
+    Task<bool> AuthorizeAsync(Guid? Id, Roles role);
+    Task<(Result result, Guid userId)> CreateUserAsync(string userName, string password);
```

Comment:
Can you make a specific model for that? Don't leave it as a genric type, please

Reply:
You mean only tuple, line 12?